packages/canary-react/src/component-tests/IcDataTable/IcDataTable.cy.tsx

```diff
@@ -1,21 +1,880 @@
+
+const DATA_TABLE_SELECTOR = "ic-data-table";
+const DEFAULT_THRESHOLD = 0.04;
+const LOADING_INDICATOR_SELECTOR = "ic-loading-indicator";
+
+export const BasicDataTable = (dataTableProps?): ReactElement => (
+  <IcDataTable
+    columns={COLS}
+    data={DATA}
+    caption="Data Tables"
+    {...dataTableProps}
+  />
+);
+
 describe("IcDataTables", () => {
+  beforeEach(() => {
+    cy.injectAxe();
+    cy.viewport(1024, 768);
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
   it("renders", () => {
-    mount(<IcDataTable columns={COLS} data={DATA} caption="Data Tables" />);
+    mount(<BasicDataTable />);
+
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+
+    cy.checkA11yWithWait();
+
+    cy.compareSnapshot({
+      name: "default",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.033),
+    });
+  });
+
+  it("should highlight row when clicked", () => {
+    mount(<BasicDataTable />);
+
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+
+    cy.findShadowEl(DATA_TABLE_SELECTOR, "tr").eq(1).click();
+    cy.findShadowEl(DATA_TABLE_SELECTOR, "tr").eq(4).click();
+
+    cy.compareSnapshot({
+      name: "highlighted-row",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.032),
+    });
+
+    cy.checkA11yWithWait();
+  });
+
+  it("should render with row headers", () => {
+    mount(
+      <IcDataTable
+        columns={ROW_HEADER_COLS}
+        data={ROW_HEADER_DATA}
+        caption="Data Tables"
+      />
+    );
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+
+    cy.checkA11yWithWait();
+
+    cy.compareSnapshot({
+      name: "with-row-headers",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.039),
+    });
+  });
+
+  it("should render embedded", () => {
+    mount(<BasicDataTable embedded />);
+
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+
+    cy.compareSnapshot({
+      name: "embedded",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.036),
+    });
+  });
+
+  it("should sort data when the sort button is clicked", () => {
+    mount(<BasicDataTable sortable />);
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+    cy.compareSnapshot({
+      name: "sortable",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.033),
+    });
+
+    cy.findShadowEl(DATA_TABLE_SELECTOR, ".sort-button").eq(1).click();
+
+    cy.compareSnapshot({
+      name: "sortable-first-sort",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.036),
+    });
+
+    cy.findShadowEl(DATA_TABLE_SELECTOR, ".sort-button").eq(2).click();
+
+    cy.compareSnapshot({
+      name: "sortable-second-sort",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_THRESHOLD + 0.034),
+    });
+  });
+
+  it('should sort data to "ascending", desending and then remove sort when the sort button is clicked three times', () => {
+    mount(<BasicDataTable sortable />);
+
+    cy.checkHydrated(DATA_TABLE_SELECTOR);
+
+    cy.findShadowEl(DATA_TABLE_SELECTOR, ".sort-button")
```

Comment:
this `cy.findShadowEl(DATA_TABLE_SELECTOR, ".sort-button").eq(0).shadow().find("ic-tooltip button")` gets repeated a lot, could that whole chain be extracted into a variable to reuse through this file?

Reply:
same with the chain that has `.eq(1)` in it, maybe two similar variables at the top of the file?

Reply:
Since the chain includes a Cypress command, it would need to be created as a Custom Command. Similar to what we have done with `cy.findShadowEl`. If this command was to be used in other tests files, I think there would be value in creating a new Custom command but as it's this single file, I'm not sure if there is much value in return. 

However, if you know where we have done something similar in another test, happy to copy that template.

Reply:
No worries, your logic makes total sense to me